src/Data/Aeson/Types/FromJSON.hs

```diff
@@ -1801,7 +1801,7 @@ instance (FromJSON1 f, FromJSON1 g, FromJSON a) => FromJSON (Product f g a) wher
 instance (FromJSON1 f, FromJSON1 g) => FromJSON1 (Sum f g) where
     liftParseJSON p pl (Object (H.toList -> [(key, value)]))
         | key == inl = InL <$> liftParseJSON p pl value <?> Key inl
-        | key == inr = InR <$> liftParseJSON p pl value <?> Key inl
+        | key == inr = InR <$> liftParseJSON p pl value <?> Key inr
```

Comment:
Wouldn't something like this be less error prone? (Change `f` and `g` to more meaningful names)

```haskell
    liftParseJSON p pl (Object (H.toList -> [(key, value)]))
        | key == inl = f InL
        | key == inr = f InR
      where
        f g = g <$> liftParseJSON p pl value <?> Key key
```